Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -692,6 +692,9 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                 throw TunnelError.startingTunnelWithoutAuthToken
             }
         } catch {
+            // Check that the error is valid and able to be re-thrown to the OS before shutting the tunnel down
+            let error = validated(error: error)
```

Comment:
This effectively removes the underlying error info from `error`.  I imagine the idea here is to catch this in the pixel we fire in line 710, but I'm hesitant to discard valuable info from that pixel.

Would you be open to:
- Moving this down to line 715, so that pixels aren't affected; and
- Optionally adding a new temporary pixel specifically to register these cases?

The downside is it's a bit more work to push a new pixel out, but the upside is we'll be able to match this pixel to the failures we're already getting, including all of the underlying error chain.

Reply:
I've updated this to only validate the error right as it's being thrown, such that the pixel call isn't affected.

I deliberately sent it with the pixel for starters so that we'd get a signal that this new error type occurred, and it would attach the original error as its underlying error, but I agree we can define something new for that instead.